### PR TITLE
Allow adapters to include python package logging in dbt logs

### LINKED ISSUE #7859: [CT-2693] [Feature] Allow debug logs for dbt-databricks
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

Add a mapping to enable `DEBUG` logs for log events from `databricks-sql-connector` similar to how `snowflake.connector` is handled today. This will assist in debuggin dbt-databricks issues.

### Describe alternatives you've considered

This must be set at the `dbt-core` level.

### Who will this benefit?

Anyone doing technical debugging of performance and behaviour of dbt-databricks.

### Are you interested in contributing this feature?

Yes. PR incoming.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by colin-rogers-dbt**
Add the ability for an adapter to mimic legacy logger and override standard dbt logging behavior to include a dependency's logging in the output. 
See dbt-snowflake PR for how it will be used: https://github.com/dbt-labs/dbt-snowflake/pull/768

resolves #7859

### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [ ] I have run this code in development and it appears to resolve the stated issue  
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

**Review by mikealfare — COMMENTED** [review]
One minor suggestion, no issues. Thanks for the references to the implementation in `dbt-snowflake`. I was about to ask then looked at the PR and saw that you already answered it.

File: core/dbt/events/event_handler.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    All logs are generated as "Note" events.
+    """
+
+    def __init__(self, event_manager: IEventManager, level):
+        super().__init__(level)
+        self.event_manager = event_manager
+
+    def emit(self, record: logging.LogRecord):
+        note = Note(msg=record.getMessage())
+        level = _log_level_to_event_level_map[record.levelno]
```
Comment (mikealfare) [inline core/dbt/events/event_handler.py:31]: The previous approach had a default scenario. Do we not need that? i.e. every value of `record.levelno` is in `_log_level_to_event_level_map` and `record.levelno` is always populated?

File: tests/unit/test_events.py [hunk 2]
```diff
@@ -83,6 +84,12 @@ def test_formatting(self):
         event = types.JinjaLogDebug(msg=[1, 2, 3])
         assert isinstance(event.msg, str)
 
+    def test_set_adapter_dependency_log_level(self):
```
Comment (mikealfare) [inline tests/unit/test_events.py:87]: There's no assertion in this test. Are we just making sure that this runs without error?

File: core/dbt/events/adapter_endpoint.py [hunk 3]
```diff
@@ -56,3 +57,10 @@ def critical(self, msg, *args):
             name=self.name, base_msg=str(msg), args=list(args), node_info=get_node_info()
         )
         fire_event(event)
+
+    @staticmethod
+    def set_adapter_dependency_log_level(package_name, level):
```
Comment (mikealfare) [inline core/dbt/events/adapter_endpoint.py:62]: I know you implemented this in `dbt-snowflake`. Do we need to do the same for `dbt-postgres`? Or is there already a default where we don't need to make any changes?

**Review by mikealfare — COMMENTED** [review]
I found a few more things that might be worth looking at. Also, I'm surprised codecov is still giving warning about missing coverage on the new code. I feel like your tests should cover it. I wonder if maybe those comments get created by the plugin, but never updated/resolved.

File: tests/unit/test_event_handler.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import logging
+
+from dbt.events.base_types import EventLevel
+from dbt.events.event_handler import DbtEventLoggingHandler, set_package_logging
+from dbt.events.eventmgr import TestEventManager
+
+
+def test_event_logging_handler_emits_records_correctly():
+    event_manager = TestEventManager()
+    handler = DbtEventLoggingHandler(event_manager=event_manager)
```
Comment (mikealfare) [inline tests/unit/test_event_handler.py:10]: It's in the test results too, but to save you the legwork, this is missing the `level` arg.

File: core/dbt/events/event_handler.py (hunk 1 above)
Comment (colin-rogers-dbt) [reply core/dbt/events/event_handler.py:31]: yes it should be populated by default

File: core/dbt/events/adapter_endpoint.py (hunk 3 above)
Comment (colin-rogers-dbt) [reply core/dbt/events/adapter_endpoint.py:62]: I don't believe there's been a need for it in dbt-postgres as of yet

File: core/dbt/events/event_handler.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    logging.DEBUG: EventLevel.DEBUG,
+    logging.INFO: EventLevel.INFO,
+    logging.WARN: EventLevel.WARN,
+    logging.WARNING: EventLevel.WARN,
+    logging.ERROR: EventLevel.ERROR,
+    logging.CRITICAL: EventLevel.ERROR,
+}
+
+
+class DbtEventLoggingHandler(logging.Handler):
```
[the remark this answers is not in the document]
Comment (colin-rogers-dbt) [reply core/dbt/events/event_handler.py:19]: I _think_ that would create a circular dependency since this depends on eventmgr which depends on logging.py

**Pull request merged**